### DISCUSSION TITLE
Option to only remove folders from OneDrive after local delete?

### DISCUSSION THREAD

**Discussion by Fire69** [abraunegg/onedrive#1448] [Q&A] [answered]
Hi,

I was using Scillion's onedrive to sync an external disk to OneDrive for some time, worked great until I upgraded Ubuntu to 20.04.  It stopped working.
I didn't look into it for several months so my disk wasn't synced anymore.  In that time I completely reorganised my photo library (~600GB) so the folder structure is now completely different.
This week I discovered and installed your version (PPA version) and I'm trying to get my disk synced again.
I started onedrive manually with --synchronize and --upload-only
My problem now is  that it's uploading almost everything again without first removing the folders that don't exist anymore locally or have been moved.
This is causing my 1TB to fill up completely.

Is there a way to let onedrive check which folders don't exist anymore locally and remove them from my Onedrive, without uploading any other changes?

Thanks!

**Comment by abraunegg** [comment, marked as answer]
@Fire69 
> I was using Scillion's onedrive to sync an external disk to OneDrive for some time, worked great until I upgraded Ubuntu to 20.04. It stopped working.

You are also lucky you do not have any major data loss or corruption. There are major bugs with that code.

> This week I discovered and installed your version (PPA version) and I'm trying to get my disk synced again.

The PPA is also out-of-date and you are running an unsupported version. You need to compile manually to any version >= v2.4.11

> My problem now is that it's uploading almost everything again without first removing the folders that don't exist anymore locally or have been moved.
This is causing my 1TB to fill up completely.
> Is there a way to let onedrive check which folders don't exist anymore locally and remove them from my Onedrive, without uploading any other changes?

What is your 'source-of-truth' ? Local disk or OneDrive?

There are several options here:
* Upload Only
* Local First
* Download Only
* Single Directory
* Dry Run

There is no real easy way to get back in sync - because of the major DB changes between the 'skilion' client, the data integrity checks and ensuring all your data is in the right place. Any 'skilion' DB representation of your sync state is incompatible - but as you have said, it already was not working for several months.

Any local folder that does not exist on OneDrive is going to be uploaded, the client has no way to understand that you do not want this to be uploaded.

I would login to OneDrive via the web interface, and move your folders around as best as possible to replicate your local directory structure - then this will be easier and not upload / download duplicate data.

**Reply by Fire69** [reply]
> You are also lucky you do not have any major data loss or corruption. There are major bugs with that code.

There was no risk, the external disk I'm syncing to Onedrive is a one-way rsync backup of my main disk. Better safe than sorry. ;)

> The PPA is also out-of-date and you are running an unsupported version. You need to compile manually to any version >= v2.4.11

I started with a compiled version but since there is no automated way to update that (correct?), I installed the PPA thinking it wouldn't be too far behind.  I'll check what version the compiled one is.

> What is your 'source-of-truth' ? Local disk or OneDrive?

Local.  Anything not on the local disk shouldn't be on OneDrive.

> Any 'skilion' DB representation of your sync state is incompatible

I started with a --resync.  If I understand correctly that drops the DB and starts from zero?

> Any local folder that does not exist on OneDrive is going to be uploaded, the client has no way to understand that you do not want this to be uploaded.

No problem, that is what I need.  I might not have explained correctly (non native English).
Let's say I had a folder 'vacation pictures' locally and that was synced to OneDrive.  I then (re)move the folder locally.
I expected onedrive to first compare all local and OneDrive folders and remove the ones from OneDrive that don't exist anymore in local.  And after that it starts uploading new local folders.
Is that how it should work?

> I would login to OneDrive via the web interface, and move your folders around as best as possible to replicate your local directory structure

I literally changed hundreds of folders, so that's not really a possibility I'm afraid.
It would probably be faster to just empty my OneDrive and start from zero :)

**Reply by abraunegg** [reply]
@Fire69 
> Local. Anything not on the local disk shouldn't be on OneDrive.

In this case use `--upload-only` so that your local state is replicated to OneDrive, however, because of client upgrade + resync, the local + remote state is unknown, thus, all files will be uploaded.

> It would probably be faster to just empty my OneDrive and start from zero :)

Potentially ... this might be the easiest solution + use the right client options so that you are not downloading data from OneDrive.

**Comment by Fire69** [comment]
I compiled a new version.
Synchronisation of my Pictures folder seemed to have gone OK, It'll take some time to check everything.

I did notice 1 (unrelated) thing.
I use a sync_list to select what folders needed to be uploaded.

sync_list contents:
```
Documents
Pictures
Videos/DV/
```
In Videos I have 2 folders, DV and DVD, only DV needs to be uploaded.
onedrive is also uploading the DVD folder.
Is my sync_list incorrect?

Thanks!

**Reply by abraunegg** [reply]
@Fire69 
This sounds like a prior bug in v2.4.10

Make sure you have stop any service + uninstalled any Ubuntu package before installing the new client as built from source.

Reference:
* https://github.com/abraunegg/onedrive/pull/1317
* https://github.com/abraunegg/onedrive/pull/1390


If you are *still* seeing an issue where you are only using the 'master' release, please open an issue ticket following the correct support process.

**Reply by Fire69** [reply]
I did some more cleaning and restarted the service.  The folder is ignored now.

Thanks for all the help!

**Reply by abraunegg** [reply]
@Fire69 
No worries. It sounds like the old version was still installed somewhere
